os/fs/procfs/fs_procfsproc.c

```diff
@@ -409,10 +410,17 @@ static ssize_t proc_entry_stat(FAR struct proc_file_s *procfile, FAR struct tcb_
 #else
 	ppid = -1;
 #endif
+	if (tcb->blink != NULL && tcb->blink->pid == getpid()) {
+		state = TSTATE_TASK_RUNNING;
+	} else if (tcb->pid == getpid()) {
+		state = TSTATE_WAIT_SEM;
+	} else {
+		state = tcb->task_state;
+	}
```

Comment:
I think you consider the TASH only. This function can be called by other thread. (No limitation) Then, this will give wrong information. How about directly checking thread name - TASH for this?